torch/distributed/fsdp/_common_utils.py

```diff
@@ -28,7 +28,7 @@
 import torch.distributed.fsdp._flat_param as flat_param_file
 import torch.nn as nn
 from torch.distributed._composable_state import _get_module_state, _State
-from torch.distributed._tensor.device_mesh import DeviceMesh
+from torch.distributed.device_mesh import DeviceMesh
```

Comment:
The old import should still be valid 🤔 : https://github.com/pytorch/pytorch/blob/f7a2bae0ac4f4c3b594619c1e39867695501735d/torch/distributed/_tensor/__init__.py#L12

Reply:
Yeah it's quite weird, this poping up some recursively imports issue in some CI and that's the motivation for the fix https://github.com/pytorch/torchtrain/actions/runs/8118646813/job/22193278282?pr=103